File: liqp-core/src/test/java/liqp/parameterized/LiquifyWithInputTest.java
package liqp.parameterized;

import static org.assertj.core.api.Assertions.assertThat;

import junitparams.JUnitParamsRunner;
import junitparams.Parameters;
import liqp.LiquidParser;
import liqp.node.LTemplate;
import org.junit.Test;
import org.junit.runner.RunWith;

@RunWith(JUnitParamsRunner.class)
public abstract class LiquifyWithInputTest {

  public abstract Object[] testParams();

  @Parameters(method = "testParams")
  @Test
  public void run(String templateString, String expectedResult, Object inputData) {
    LTemplate template = LiquidParser.newInstance().parse(templateString);
    String rendered = String.valueOf(template.render(inputData));

    assertThat(rendered).isEqualTo(expectedResult);
  }
}

File: liqp-core/src/test/kotlin/liqp/parameterized/LiquifyWithInputTest.kt
package liqp.parameterized

import assertk.assert
import assertk.assertions.isEqualTo
import liqp.createTestParser
import liqp.parseIfNecessary
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.Parameterized

@RunWith(Parameterized::class)
abstract class LiquifyWithInputTest(val templateString: String, val expectedResult: String?, val inputData: Any? = null) {
  @Test
  open fun run() {
    val template = createTestParser {}.parse(templateString)
    val input = try {
      inputData.parseIfNecessary()
    } catch (e: Exception) {
      inputData
    }
    val rendered = template.render(input)
    assert(rendered).isEqualTo(expectedResult)
  }
}
